tchannel-core/src/main/java/com/uber/tchannel/schemes/JSONSerializer.java

```diff
@@ -47,7 +48,8 @@ public String decodeEndpoint(ByteBuf arg1) {
     public Map<String, String> decodeHeaders(ByteBuf arg2) {
         String headerJSON = arg2.toString(CharsetUtil.UTF_8);
         arg2.release();
-        return new Gson().fromJson(headerJSON, HEADER_TYPE);
+        Map<String, String> headers = new Gson().fromJson(headerJSON, HEADER_TYPE);
```

Comment:
It can be a variety of things, and it depends on the caller. This prevents a NPE when Gson can't decode invalid json e.g. empty strings or nulls.